runtimebp/signal.go

```diff
@@ -17,6 +19,30 @@ var defaultSignals = []os.Signal{
 	syscall.SIGTERM,
 }
 
+// State is an enum representing the runtime state of the baseplate server
+type State int
+
+const (
+	StateUnknown State = iota
+	StateRunning
+	StateShuttingDown
+)
+
+// drainer is used to track the server state
+//
+// This drainer is updated by HandleShutdown before invoking the
+// user supplied ShutdownHandler
+var drainer = baseplate.Drainer()
+
+// ServerState returns the current runtimebp.State of the application
+func ServerState() State {
+	if drainer.IsHealthy(context.Background()) {
+		return StateRunning
+	}
+
+	return StateShuttingDown
+}
```

Comment:
I don't really think this should be here like this, to me `HandleShutdown` should just be helper and this state should be managed in `baseplate.Serve` on the passed in `Baseplate` object rather than as a global.